MDES 3.2 Operational Enumerator (#3468)

rsutphin opened the pull request:
Operational enumerator for MDES 3.2. It's exactly the same as the MDES 3.1 one because there are no operational data changes between those two versions.

rsutphin committed:
Switch back to released version of ncs_navigator_configuration.

The latest released version contains the commit that was previously
locked.

(In the future, please note why you are using the git version in a
comment in the Gemfile.)

rsutphin committed:
Add MDES 3.2 operational enumerator. #3468.

It's identical to the 3.1 one because there are no operational data
changes for MDES 3.2.

jpatel commented:
When I was looking for the differences for the Ops, I found 3 additional variables in the Provider Table for MDES 3.2. I was looking into xls sheet. 
1) MEDICAL_RECORDS - MEDICAL_RECORDS_CL1- required
2) NM_PRENATAL_VISITS - non_mdes_coded_field - not_required
3) NM_FIRST_PRENATAL_VISITS - non_mdes_coded_field - not_required

rsutphin commented:
That is a good point. I was using @phebal's [migration analysis](https://code.bioinformatics.northwestern.edu/issues/issues/show/3630), but of course the migration doesn't need to concern itself with entirely new fields. Diffing the provider table does show those new variables:

```
> pp $mdes31['provider'].diff $mdes32['provider']
#<NcsNavigator::Mdes::Differences::Entry:0x007ff8982904a8
 @attribute_differences=
  {:variables=>
    #<NcsNavigator::Mdes::Differences::Collection:0x007ff8982904d0
     @entry_differences=
      {"provider_info_source"=>
        #<NcsNavigator::Mdes::Differences::Entry:0x007ff898287f88
         @attribute_differences=
          {:type=>
            #<NcsNavigator::Mdes::Differences::Entry:0x007ff898288258
             @attribute_differences=
              {:name=>
                #<struct NcsNavigator::Mdes::Differences::Value
                 left="information_source_cl2",
                 right="information_source_cl9">,
               :code_list_by_value=>
                #<NcsNavigator::Mdes::Differences::Collection:0x007ff89827c318
                 @entry_differences={},
                 @left_only=[],
                 @right_only=["-7"]>,
               :code_list_by_label=>
                #<NcsNavigator::Mdes::Differences::Collection:0x007ff898288280
                 @entry_differences={},
                 @left_only=[],
                 @right_only=["not applicable"]>}>}>},
     @left_only=[],
     @right_only=
      ["medical_records", "nm_prenatal_visits", "nm_first_prenatal_visits"]>}>
```

Cases does not support these attributes yet, so they can't be enumerated out of it. `nm_prenatal_visits` and `nm_first_prenatal_visits` are optional, so we can leave those aside. `medical_records` is required, so I'll need to provide a value there.

rsutphin committed:
Hard-code PROVIDER.MEDICAL_RECORDS to -4. #3468.

This is a new mandatory field that Cases does not support collecting
yet (#3632).

rsutphin commented:
Enumerator updated. See also [Cases-3632](https://code.bioinformatics.northwestern.edu/issues/issues/show/3632).
